risc0/zkvm/src/receipt.rs

```diff
@@ -112,8 +113,16 @@ where
         return Ok(());
     }
 
-    let check_code = |_po2: u32, _root: &Digest| -> Result<(), VerificationError> {
-        // TODO
+    let control_id = ControlId::new();
```

Comment:
Would it be possible to call compute_control_id() from within this context to prevent preforming code gen? Alternatively would it be possible to call it when the loader if first run, then place the results into the Receipt?

It seems like having these values are generated by the host running the code and required by the host verifying the code, so I assume we should include it in the Receipt right? What happens when a prover / verifier have a different set of control_id's?